app/controllers/books_controller.rb

```diff
@@ -62,15 +69,21 @@ def filter_books
   end
 
   def category
-    Category.find_by(name: permitted_params[:category_name])
+    @book = Book.find(params[:id])
+    @category = Category.find_by(name: permitted_params[:category_name])
   end
 
   def load_books
+    @user = current_user
```

Comment:
Metoda nazywa się `load_books`, ale oprócz `@books` przypisuje też obiekt `@user`, to jest niespójne. Jeśli musisz mieć załadowany obiekt usera, wydziel osobne before_action do tego celu.